### PR TITLE
Solve alpha bug for Developer Guide

### LINKED ISSUE #221: DG - Example command format not updated
image
At the design architecture part, one example execution is given and the command format for delete student is not updated as deleteStu for both the paragraphs and sequence diagram.

### LINKED ISSUE #226: DG - Example command wrong format
image
The example command in Logic Component, API: Logic.java part remains as delete 1 instead of deleteStu 1

### LINKED ISSUE #228: DG - Storage Inaccuracies
image
The first point says can save address book data and user preference data.
Maybe can include assignments and tutorials as they are in separate json. 
Also Im not very sure if the name addressbook data is appropriate? Are we sticking to addressbook or should we change it to student data?

### LINKED ISSUE #230: DG - Feature Architecture Design
image
Might be a bit redundant but I feel like adding some margin on the left of the dots looks visually better hahahhaa.

### LINKED ISSUE #237: DG - Incorrect Activity Diagram
image
The above activity diagram has some issues:
1. Conditions are inside the diamond, which is not the convention used in CS2103T
2. Outcome of the conditions are not wrapped inside []
3. Perhaps add student command entered is not that direct, can change to active form, enter add student command.

### LINKED ISSUE #238: DG - Class diagram error
image
Under the add feature, the class diagram shows composition using new classes, but also write it down under the attributes section. This duplicated section is not allowed in CS2103T

### LINKED ISSUE #244: DG - Inconsistent bold text in glossary
image
Some texts are bolded, while some are not, which is inconsistent

### PR DESCRIPTION
**Pull request by SamsonChew**
- [x] Change Architecture Sequence Diagram (221)
- [x] Change Delete Sequence Diagram (226)
- [x] Solve Storage component inconsistency (228)
- [x] Improve feature architecture design paragraph (230)
- [x] Correct Activity feature activity diagram (237)
- [x] Correct Activity feature class diagram (238)
- [x] Solve inconsistency in glossary section (244)

closes #221 
closes #226 
closes #228 
closes #230 
closes #237 
closes #238 
closes #244

### CODE REVIEW

File: docs/diagrams/DeleteSequenceDiagram.puml [hunk 1]
```diff
@@ -14,10 +14,10 @@ box Model MODEL_COLOR_T1
 participant "model:Model" as Model MODEL_COLOR
 end box
 
-[-> LogicManager : execute("delete 1")
+[-> LogicManager : execute("deleteStu 1")
 activate LogicManager
 
-LogicManager -> AddressBookParser : parseCommand("delete 1")
+LogicManager -> AddressBookParser : parseCommand("deleteStu 1")
```
Comment (tansanchian) [inline docs/diagrams/DeleteSequenceDiagram.puml:20]: Good to see the command words are now consistent across the documentation.

**Review by tansanchian — APPROVED** [review]
LGTM!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -621,16 +621,16 @@ TO BE CHANGED
...
+| **NUS**        | National University of Singapore , A University Located in Central Singapore                                              |
+| **TA**         | Teaching Assistant, a tutor who will conduct tutorials in National University of Singapore                                |
+| **SoC**        | School of Computing in National University of Singapore                                                                   |
 | **Student ID** | A particular format of ID followed in NUS, Starting with a Alphabet followed by 7 digits and ending with an Alphabet e.g A0252195L |
-| **GUI**        | Graphical User Interface , all the part of the application which you can interact with your mouse                                  |
-| **CLI**        | Command Line Interface , part of application which can only be used with commands from the Keyboard                                |
+| **GUI**        | Graphical User Interface , all the part of the application which you can interact with your mouse                         |
+| **CLI**        | Command Line Interface , part of application which can only be used with commands from the Keyboard                       |
```
Comment (waihongteh) [inline docs/UserGuide.md:633]: Consistent text format. Nice!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -166,27 +167,27 @@ This section describes some noteworthy details on how certain features are imple
...
+    * **Challenge**: It requires careful handling of the parsing rules to ensure correctness since the parser is a crucial layer between user input and the system’s execution.
 
 2. **Avoiding Duplicates with Tutorial and Student Validation**:
-We ensure that duplicate students or non-existent tutorials are caught early in ModelManager. This provides immediate feedback and prevents inconsistent states in the data model.
+   We ensure that duplicate students or non-existent tutorials are caught early in `ModelManager`. This provides immediate feedback and prevents inconsistent states in the data model.
 
-* **Benefit**: Validation at the model level ensures the integrity of the data.
-* **Challenge**: This requires checks across multiple classes (such as TutorialList and Student) to ensure consistency without impacting performance.
+    * **Benefit**: Validation at the model level ensures the integrity of the data.
+    * **Challenge**: This requires checks across multiple classes (such as `TutorialList` and `Student`) to ensure consistency without impacting performance.
```
Comment (waihongteh) [inline docs/DeveloperGuide.md:182]: Good to see that you take care of mall details like margin too!

**Review by waihongteh — APPROVED** [review]
LGTM! Feel free to merge!

**Pull request merged**